sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -237,15 +237,26 @@ def receive(self, batch=False, max_batch_size=300, max_wait_time=None):
                     if isinstance(exception, ImportError):
                         raise exception
                     self._amqp_transport.check_link_stolen(self, exception)
+                    if self._last_received_event:
+                        self._offset = self._last_received_event.offset
+
+                    if self._is_non_retryable(exception):
+                        last_exception = self._handle_exception(exception)
+                        _LOGGER.info(
+                            "% operation has encountered an exception: %r.",
```

Comment:
would it be better to have the following message
```suggestion
                            "% operation has encountered a non-retryable exception: %r.",
```

Reply:
I dont think we need to mention it being non-retryable. I dont see us making at distinction anywhere else ( outside of messaging too )

Reply:
leaving as is, since we don't want to confirm that it's non-retryable yet. + It's retrying from the event processor, so it may be a confusing message.